doc-surrealdb_versioned_docs/version-2.x/reference-guide/security-best-practices.mdx

```diff
@@ -0,0 +1,323 @@
+
+**Example**
+
+```surql
+
+RETURN string::html::sanitize("<h1>Safe Title</h1><script>alert('XSS')</script><p>Safe paragraph. Not safe <span onload='logout()'>event</span>.</p>");
+
+['<h1>Safe Title</h1><p>Safe paragraph. Not safe <span>event</span>.</p>']
+```
+
+## JSON Web Tokens
+
+When configuring how [JSON Web Tokens](https://datatracker.ietf.org/doc/html/rfc7519) are verified before authenticating a [system](https://surrealdb.com/docs/surrealdb/2.x/security/authentication#system-users) or [record](https://surrealdb.com/docs/surrealdb/2.x/security/authentication#record-users) user with [`DEFINE ACCESS ... TYPE JWT`](https://surrealdb.com/docs/surrealdb/2.x/surrealql/statements/define/access/jwt) or [`DEFINE ACCESS ... TYPE RECORD ... WITH JWT`](https://surrealdb.com/docs/surrealdb/2.x/surrealql/statements/define/access/record#with-json-web-token), we recommend using an asymmetric algorithm (i.e. `PSXXX`,  `RSXXX` , `ECXXX`) when only a mechanism for token verification is being defined. This ensures that the only key stored by SurrealDB is a public key that does not represent a threat in the event of a compromise. On the other hand, symmetric algorithms (i.e. `HSXXX`) use the same key for signature and verification, which the attacker would be able to use to issue tokens that would be trusted by SurrealDB.
+
+**Example**
+
+```surql
+DEFINE ACCESS token ON DATABASE TYPE RECORD WITH JWT
+ALGORITHM RS256 KEY "-----BEGIN PUBLIC KEY-----
+MUO52Me9HEB4ZyU+7xmDpnixzA/CUE7kyUuE0b7t38oCh+sQouREqIjLwgHhFdhh3cQAwr6GH07D
+ThioYrZL8xATJ3Youyj8C45QnZcGUif5PkpWXDi0HJSoMFekbW6Pr4xuqIqb2LGxGDVJcLZwJ2AS
+Gtu2UAfPXbBD3ffiad393M22g1iHM80YaNi+xgswG7qtXE4lR/Lt4s0MeKKX7stdWI1VIsoB+y3i
+r/OWUvJPjjDNbAsyy8tQmxydv+FUnLEP9TNT4AhN4DXcJ+XsDtW7OWt4EdSVDeKpGbIMvIrh1Pe+
+Nilj8UHNyNDHa2AjK3seMo6CMvaIQJKj5o4xGFblFGwvvPD03SbuQLs1FdRjsZCeWLdYeQ3JDHE9
+sFG7DCXlpMJcaYT1mf4XHJ0gPekNLQyewTY3Vxf7FgV3GCNjV20kcDFgJA2+iVW2wSrb+txD1ycE
+kbi8jh0pedWwE40VQWaTh/8eAvX7IHWya/AEro25mq+m6vktNZLbvLphhp586kJK3Tdt3YjpkPre
+M3nkFWOWurIyKbtIV9JemfwCgt89sNV45dTlnEDEZFFGnIgDnWgx3CUo4XmhICEQU8+tklw9jJYx
+iCTjhbIDEBHySSSc/pQ4ftHQmhToTlQeOdEy4LYiaEIgl1X+hzRH1hBYvWlNKe4EY1nMCKcjgt0=
+-----END PUBLIC KEY-----";
+```
+
+Additionally, we recommend using [JSON Web Key Sets](https://datatracker.ietf.org/doc/html/rfc7517) to configure the verification algorithm and key from a remote authoritative source using the `URL` clause instead of providing them directly to SurrealDB using the `ALGORITHM` and `KEY` clauses. This ensures that the original token issuer will be able to rotate keys in the event of a compromise to prevent potentially compromised tokens to be used with your application without affecting the availability of your service.
+
+**Example**
```

Comment:
```suggestion
### Example: Define a JWT access with a JWKS
```

Reply:
I would just say "Define JWT access with JWKS" or "Define a JWT access method with JWKS".

Reply:
Addressed in [d56a22c](https://github.com/surrealdb/docs.surrealdb.com/pull/651/commits/d56a22c380c2d35275ec64ebc264ac3c1f21e194).